app/src/main/java/org/projectbuendia/client/net/JsonErrorListener.java

```diff
@@ -0,0 +1,70 @@
+// Copyright 2015 The Project Buendia Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License.  You may obtain a copy
+// of the License at: http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distrib-
+// uted under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+// OR CONDITIONS OF ANY KIND, either express or implied.  See the License for
+// specific language governing permissions and limitations under the License.
+
+package org.projectbuendia.client.net;
+
+import android.graphics.Color;
+import android.graphics.PorterDuff;
+import android.graphics.drawable.NinePatchDrawable;
+import android.widget.Toast;
+
+import com.android.volley.Response.ErrorListener;
+import com.android.volley.VolleyError;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonParser;
+
+import org.projectbuendia.client.App;
+
+public class JsonErrorListener implements ErrorListener {
```

Comment:
@capnfabs I added a TODO to refactor the message using eventbus and snackbar.
I renamed the class to OpenMrsErrorListener.